framework/BaseYii.php

```diff
@@ -441,12 +441,29 @@ public static function info($message, $category = 'application')
      *     \Yii::endProfile('block2');
      * \Yii::endProfile('block1');
      * ```
+     *
+     * Profiler is configured as [[yii\base\Profiler]] by default. You can override it
+     * via application components configuration as follows:
+     *
+     * ```php
+     * 'components' => [
+     *     'profiler' => [
+     *         'class' => 'your\own\Profiler'
+     *     ],
+     *     ...
+     * ]
+     * ```
+     *
      * @param string $token token for the code block
      * @param string $category the category of this log message
      * @see endProfile()
      */
     public static function beginProfile($token, $category = 'application')
     {
+        if (static::$app !== null) {
+            return static::$app->getProfiler()->begin($token, $category);
```

Comment:
Please, don't use such return syntax for methods that returns void - it is really confusing.